src/libexpr/primops.cc

```diff
@@ -3767,6 +3778,68 @@ static RegisterPrimOp primop_hashString({
     .fun = prim_hashString,
 });
 
+static void prim_convertHash(EvalState & state, const PosIdx pos, Value * * args, Value & v)
+{
+    state.forceAttrs(*args[0], pos, "while evaluating the first argument passed to builtins.convertHash");
+    auto &inputAttrs = args[0]->attrs;
+
+    Bindings::iterator iteratorHash = getAttr(state, state.symbols.create("hash"), inputAttrs, "while locating the attribute 'hash'");
+    auto hash = state.forceStringNoCtx(*iteratorHash->value, pos, "while evaluating the attribute 'hash'");
+
+    Bindings::iterator iteratorHashAlgo = getExistingAttr(state, state.symbols.create("hashAlgo"), inputAttrs, "while locating the attribute 'hashAlgo'");
+    std::optional<HashType> ht = std::nullopt;
+    if (iteratorHashAlgo != inputAttrs->end()) {
+        ht = parseHashType(state.forceStringNoCtx(*iteratorHashAlgo->value, pos, "while evaluating the attribute 'hashAlgo'"));
+    }
+
+    Bindings::iterator iteratorToHashFormat = getAttr(state, state.symbols.create("toHashFormat"), args[0]->attrs, "while locating the attribute 'toHashFormat'");
+    Base hb = parseHashBase(state.forceStringNoCtx(*iteratorToHashFormat->value, pos, "while evaluating the attribute 'toHashFormat'"));
+
+    v.mkString(Hash::parseAny(hash, ht).to_string(hb, hb == SRI));
+}
+
+static RegisterPrimOp primop_convertHash({
+    .name = "__convertHash",
+    .args = {"args"},
+    .doc = R"(
+      Return the specified representation of a hash string. e.g.
+
+      ```nix
+      # This returns "sha256-47DEQpj8HBSa+/TImW+5JCeuQeRkm5NMpJWZG3hSuFU="
+      builtins.convertHash {
+        hash = "e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855";
+        toHashFormat = "sri";
+        hashAlgo = "sha256";
+      }
+      ```
+
+      ```nix
+      # This returns "e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855"
+      builtins.convertHash {
+        hash = "sha256-47DEQpj8HBSa+/TImW+5JCeuQeRkm5NMpJWZG3hSuFU=";
+        toHashFormat = "base16";
+      }
+      ```
+
+      ```nix
+      # This returns "sha256-47DEQpj8HBSa+/TImW+5JCeuQeRkm5NMpJWZG3hSuFU="
+      builtins.convertHash {
+        hash = "sha256:e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855";
+        toHashFormat = "sri";
+      }
+      ```
+
+      The result hash is the *toHashFormat* representation of the hash *hash*.
+
+      The hash algorithm specified by *hashAlgo* must be one of `"md5"`,
+      `"sha1"`, `"sha256"` or `"sha512"`. The hash format specified by
+      *toHashFormat* must be one of `"base16"`, `"base32"`, `"base64"` or `"sri"`.
+      The attribute `"hashAlgo"` may be omitted when *hash* is an SRI hash or is
+      prefixed with the hash algorithm followed by a colon (`"${hashAlgo}:"`).
```

Comment:
```suggestion
      - `hash`
      
        The hash to be converted.
        The hash format is detected automatically.

      - `hashAlgo`
      
        The algorithm used to create the hash. Must be one of
        - `"md5"`
        - `"sha1"`
        - `"sha256"`
        - `"sha512"`
        
        The attribute may be omitted when `hash` is an [SRI hash](https://www.w3.org/TR/SRI/#the-integrity-attribute) or when the hash is prefixed with the hash algorithm name followed by a colon.
        That  `<hashAlgo>:<hashBody>` syntax is supported for backwards compatibility with existing tooling.
      
      - `toHashFormat`
      
        The format of the resulting hash. Must be one of
        - `"base16"`
        - `"base32"`
        -  `"base64"`
        - `"sri"`
```

Please move that to the beginning, after the summary sentence.